### PR TITLE
Adds controllers for cards.

### PR DESCRIPTION
**Pull request by degoeym**
Placeholder PR for creation of controllers and actions.

### CODE REVIEW

File: FiveRingsDb/Controllers/CardRulingsController.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+
+namespace FiveRingsDb.Controllers
+{
+    public class CardRulingsController : ControllerBase
+    {
+        [HttpGet("{id}")]
```
Comment (fatihi) [inline FiveRingsDb/Controllers/CardRulingsController.cs]:
```suggestion
        [HttpGet]
```

File: FiveRingsDb/Controllers/CardRulingsController.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            return Ok("Not yet implemented");
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetRuling()
+        {
+            return Ok("Not yet implemented");
+        }
+
+        [HttpDelete]
```
Comment (fatihi) [inline FiveRingsDb/Controllers/CardRulingsController.cs]:
```suggestion
        [HttpDelete("{id}")]
```

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs [hunk 3]
```diff
@@ -6,19 +6,19 @@ namespace FiveRingsDb.Tests.Controllers
 {
     public class CardsControllerTests
     {
-        private CardsController sut;
```
Comment (Fuzzylog1cza) [inline FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: Sut stands for subject under test and is a well understood pattern for testing. Reason behind it is at a glance you know what is being tested.

I'd prefer the rename was changed back.

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs [hunk 4]
```diff
@@ -6,19 +6,19 @@ namespace FiveRingsDb.Tests.Controllers
...
+            cardsController = new CardsController();
         }
 
         [Test]
         public void Get_ReturnsValues()
         {
-            var result = sut.Get();
-            result.Value.Should().BeEquivalentTo(new string[] { "value1", "value2" });
+            var result = cardsController.GetCard();
+            result.Should().BeEquivalentTo(new string[] { "Not yet implemented" });
```
Comment (Fuzzylog1cza) [inline FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: This assertion should rather check that the status code is ok and the message value as not implanted

File: FiveRingsDb/Controllers/CardRulingsController.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.AspNetCore.Mvc;
+
+namespace FiveRingsDb.Controllers
+{
+    public class CardRulingsController : ControllerBase
+    {
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetRulings()
+        {
+            return Ok("Not yet implemented");
```
Comment (Fuzzylog1cza) [inline FiveRingsDb/Controllers/CardRulingsController.cs]: If it's not implemented we should rather use an error result.

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs (hunk 3 above)
Comment (degoeym) [reply FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: Actually did not know this. I'll fix it momentarily.

File: FiveRingsDb/Controllers/CardRulingsController.cs (hunk 1 above)
Comment (degoeym) [reply FiveRingsDb/Controllers/CardRulingsController.cs]: Should be fixed now.

File: FiveRingsDb/Controllers/CardRulingsController.cs (hunk 2 above)
Comment (degoeym) [reply FiveRingsDb/Controllers/CardRulingsController.cs]: Fixed.

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs (hunk 3 above)
Comment (degoeym) [reply FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: Fixed.

File: FiveRingsDb/Controllers/CardRulingsController.cs (hunk 5 above)
Comment (Fuzzylog1cza) [reply FiveRingsDb/Controllers/CardRulingsController.cs]: Ah, that makes sense thanks.

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs [hunk 6]
```diff
@@ -17,8 +18,8 @@ public void Setup()
         [Test]
         public void Get_ReturnsValues()
         {
-            var result = sut.Get();
-            result.Value.Should().BeEquivalentTo(new string[] { "value1", "value2" });
+            var result = sut.GetCard();
```
Comment (fatihi) [inline FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: Should be GetCards

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs [hunk 7]
```diff
@@ -17,8 +18,8 @@ public void Setup()
...
+            result?.Status.Should().Be(TaskStatus.RanToCompletion);
```
Comment (fatihi) [inline FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: Will this not pass if GetCards() returns null?

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs (hunk 6 above)
Comment (degoeym) [reply FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: I updated this test to be `GetCard_ReturnsValues()` and added another test specific to `GetCards`.

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs (hunk 4 above)
Comment (degoeym) [reply FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: The methods themselves have been updated to return a 204 No Content call now, and the tests have been updated to simply make sure they got task back that ran to completion. Really these are just dummy placeholders at the moment.

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs [hunk 8]
```diff
@@ -1,3 +1,8 @@
+using FiveRingsDb.Controllers;
```
Comment (programming-wolf) [inline FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: Can we please put the `using` declarations inside the namespace definition? That's good practice and also recommended by microsoft. It limits the usings to the namespace.

File: FiveRingsDb/Controllers/CardRulingsController.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+{
+    public class CardRulingsController : ControllerBase
+    {
+        [HttpGet]
+        public async Task<IActionResult> GetRulings()
+        {
+            return NoContent();
+        }
+    }
+}
```
Comment (programming-wolf) [inline FiveRingsDb/Controllers/CardRulingsController.cs]: Oh, and we should end all our files with a new line :)

File: FiveRingsDb/Controllers/CardsController.cs [hunk 10]
```diff
@@ -11,9 +14,63 @@
...
+        }
+
+        [HttpGet("{id}/rulings")]
+        public async Task<IActionResult> GetCardRulings(string id)
+        {
+            return NoContent();
+        }
+
+        [HttpPost("{id}/rulings")]
+        public async Task<IActionResult> AddCardRuling(string id)
```
Comment (programming-wolf) [inline FiveRingsDb/Controllers/CardsController.cs]: Is this an API method that we want to provide to the public? Without any authentication/authorization?

Comment (programming-wolf) [inline FiveRingsDb/Controllers/CardsController.cs]: I think the object is missing, right? The ruling string?

File: FiveRingsDb/Controllers/CardsController.cs [hunk 11]
```diff
@@ -11,9 +14,63 @@
...
+        }
+
+        [HttpGet("{id}/reviews")]
+        public async Task<IActionResult> GetReviews(string id)
+        {
+            return NoContent();
+        }
+
+        [HttpPost("{id}/reviews")]
+        public async Task<IActionResult> AddReview(string id)
```
Comment (programming-wolf) [inline FiveRingsDb/Controllers/CardsController.cs]: Same as for the ruling above.

File: FiveRingsDb.Tests/Controllers/CardsControllerTests.cs (hunk 8 above)
Comment (degoeym) [reply FiveRingsDb.Tests/Controllers/CardsControllerTests.cs]: @programming-wolf Do you have a source on Microsoft suggesting this? Honestly the first I've ever heard of it and to that end it just seems weird to me.

File: FiveRingsDb/GlobalSuppressions.cs [hunk 12]
```diff
@@ -0,0 +1,6 @@
+﻿// This file is used by Code Analysis to maintain SuppressMessage 
+// attributes that are applied to this project.
+// Project-level suppressions either have no target or are given 
+// a specific target and scoped to a namespace, type, member, etc.
+
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.OrderingRules", "SA1200:Using directives must be placed correctly", Justification = "<Pending>")]
```
Comment (programming-wolf) [inline FiveRingsDb/GlobalSuppressions.cs]: Sneaky 😏

**Pull request merged**